lib/resourceapply/batch.go

```diff
@@ -29,16 +29,56 @@ func ApplyJobv1(ctx context.Context, client batchclientv1.JobsGetter, required *
 		return nil, false, nil
 	}
 
+	var original batchv1.Job
+	existing.DeepCopyInto(&original)
 	modified := pointer.BoolPtr(false)
 	resourcemerge.EnsureJob(modified, existing, *required)
 	if !*modified {
 		return existing, false, nil
 	}
-
 	if reconciling {
-		klog.V(2).Infof("Updating Job %s/%s due to diff: %v", required.Namespace, required.Name, cmp.Diff(existing, required))
+		if diff := cmp.Diff(&original, existing); diff != "" {
+			klog.V(2).Infof("Updating Job %s/%s due to diff: %v", required.Namespace, required.Name, diff)
+		} else {
+			klog.V(2).Infof("Updating Job %s/%s with empty diff: possible hotloop after wrong comparison", required.Namespace, required.Name)
+		}
 	}
 
 	actual, err := client.Jobs(required.Namespace).Update(ctx, existing, metav1.UpdateOptions{})
 	return actual, true, err
 }
+
+// ApplyCronJobv1 applies the required CronJob to the cluster.
+func ApplyCronJobv1(ctx context.Context, client batchclientv1.CronJobsGetter, required *batchv1.CronJob, reconciling bool) (*batchv1.CronJob, bool, error) {
+	existing, err := client.CronJobs(required.Namespace).Get(ctx, required.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		klog.V(2).Infof("CronJob %s/%s not found, creating", required.Namespace, required.Name)
+		actual, err := client.CronJobs(required.Namespace).Create(ctx, required, metav1.CreateOptions{})
+		return actual, true, err
+	}
+	if err != nil {
+		return nil, false, err
+	}
+	// if we only create this resource, we have no need to continue further
+	if IsCreateOnly(required) {
+		return nil, false, nil
+	}
+
+	var original batchv1.CronJob
+	existing.DeepCopyInto(&original)
+	modified := pointer.BoolPtr(false)
+	resourcemerge.EnsureCronJob(modified, existing, *required)
+	if !*modified {
+		return existing, false, nil
+	}
+	if reconciling {
+		if diff := cmp.Diff(&original, existing); diff != "" {
+			klog.V(2).Infof("Updating CronJob %s/%s due to diff: %v", required.Namespace, required.Name, diff)
+		} else {
+			klog.V(2).Infof("Updating CronJob %s/%s with empty diff: possible hotloop after wrong comparison", required.Namespace, required.Name)
+		}
+	}
```

Comment:
Reading the copy of this code for fifth time makes me think whether we can somehow DRY these `Apply*` methods out :/ Not an actual suggestion for you to fix, just thinking loud.

Reply:
Yes, that would be great :( I will create a card so we can keep track of these potential places to be refactored. This is a good idea for a shift-week activity or for anyone interested in helping to develop the CVO.